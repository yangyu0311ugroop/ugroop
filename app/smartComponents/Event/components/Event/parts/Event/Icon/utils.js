/**
 * Created by stephenkarpinskyj on 20/8/18.
 */

import React from 'react';
import GridContainer from 'components/GridContainer';
import GridItem from 'components/GridItem';
import { EventIcon } from 'viewComponents/Event';
import { EVENT_HELPERS } from 'utils/helpers/events';

const makeIconOptions = (type, subtype) => {
  const icons = EVENT_HELPERS.getEventIcons(type, subtype);
  return icons.reduce((acc, i) => {
    if (i.hidden) return acc;
    return [
      ...acc,
      {
        value: i.type,
        children: (
          <GridContainer alignItems="center">
            <GridItem>
              <EventIcon
                type={type}
                subtype={subtype}
                iconOverride={i.type}
                size="base"
              />
            </GridItem>
            <GridItem>{i.name}</GridItem>
          </GridContainer>
        ),
      },
    ];
  }, []);
};

const iconOptions = (type, subtype) => {
  const icons = EVENT_HELPERS.getEventIcons(type, subtype);
  return icons.reduce((acc, i) => {
    if (i.hidden) return acc;
    return [
      ...acc,
      {
        ...i,
        value: i.type,
        children: i.name,
      },
    ];
  }, []);
};

export default {
  makeIconOptions,
  iconOptions,
};
