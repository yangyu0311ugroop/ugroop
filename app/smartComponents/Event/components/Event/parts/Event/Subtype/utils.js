/**
 * Created by stephenkarpinskyj on 20/8/18.
 */

import React from 'react';
import { EVENT_HELPERS } from 'utils/helpers/events';
import { H5 } from 'viewComponents/Typography';
import { EventHeading } from 'viewComponents/Event';

const makeSubtypeOptions = (type, filter = null) => {
  const subtypes = EVENT_HELPERS.getEventTypeConstants(type).getTypes();
  const options = Object.values(subtypes).map(({ type: subtype, name }) => ({
    value: subtype,
    children: <EventHeading Typography={H5}>{name}</EventHeading>,
  }));

  return filter ? options.filter(filter) : options;
};

export default {
  makeSubtypeOptions,
};
