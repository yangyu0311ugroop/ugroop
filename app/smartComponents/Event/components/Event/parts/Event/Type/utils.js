/**
 * Created by stephenkarpinskyj on 20/8/18.
 */

import React from 'react';
import { EVENT_CONSTANTS } from 'utils/constants/events';
import { H5 } from 'viewComponents/Typography';
import { EventHeading } from 'viewComponents/Event';

const renderTypeOptions = () =>
  Object.values(EVENT_CONSTANTS.EVENTS).map(({ type, name }) => ({
    value: type,
    children: <EventHeading Typography={H5}>{name}</EventHeading>,
  }));

export default {
  renderTypeOptions,
};
