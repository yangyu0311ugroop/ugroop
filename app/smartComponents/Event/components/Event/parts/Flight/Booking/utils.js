/**
 * Created by stephenkarpinskyj on 20/8/18.
 */

import React from 'react';
import _ from 'lodash';
import { EVENT_CONSTANTS } from 'utils/constants/events';
import P from 'viewComponents/Typography';
import FlightBooking from 'smartComponents/Event/components/FlightBooking/layouts';

export const EMPTY = '0';
export const CREATE_NEW = 'new';

const findBookingId = (bookingIds, id) => {
  const idNumber = Number.parseInt(id, 10);
  return _.find(bookingIds, currentId => idNumber === currentId);
};

const renderBooking = dataId => (
  <FlightBooking dataId={dataId} variant={EVENT_CONSTANTS.VARIANTS.option} />
);

const renderBookingOptions = (bookingIds, hideAddNew) => {
  const options = [];
  const push = (id, children) => {
    options.push({ value: id.toString(), children });
  };

  push(EMPTY, <P noMargin>No flight itinerary</P>);

  if (bookingIds) {
    bookingIds.forEach(id => {
      push(id, renderBooking(id));
    });
  }

  if (hideAddNew) return options;

  push(
    CREATE_NEW,
    <P noMargin weight="bold">
      Add new itinerary
    </P>,
  );

  return options;
};

export default {
  findBookingId,
  renderBooking,
  renderBookingOptions,
};
