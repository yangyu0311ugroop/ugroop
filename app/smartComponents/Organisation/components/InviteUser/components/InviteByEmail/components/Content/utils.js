const inviteYourself = me => (_, value) =>
  value !== me ? true : "Hey that's you.";
const inviteOwner = (owner, people) => (_, value) =>
  value !== owner && people.indexOf(value) === -1
    ? true
    : 'This person is already a member.';
const invitePending = (owner, people) => (_, value) =>
  people.indexOf(value) === -1 ? true : 'This person has a pending invitation.';
export default {
  inviteYourself,
  inviteOwner,
  invitePending,
};
