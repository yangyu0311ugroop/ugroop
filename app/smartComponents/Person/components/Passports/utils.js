import has from 'lodash/has';
import omit from 'lodash/omit';

export const validateData = data => {
  let formData = data;
  if (has(data, 'passportOtherType')) {
    formData.passportType = data.passportOtherType;
    formData = omit(formData, ['passportOtherType']);
  }
  return formData;
};

export const PASSPORT_UTILS = {
  validateData,
};
