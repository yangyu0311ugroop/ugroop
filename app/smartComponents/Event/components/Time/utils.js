/**
 * Created by stephenkarpinskyj on 4/9/18.
 */

import _ from 'lodash';
import { dateDisplay } from 'utils/constant';
import { pluralizeText } from 'utils/stringAdditions';
import { FIRST_DATE } from 'utils/constants/dateTime';
import { MOMENT_HELPERS } from 'utils/helpers/moment';
import { NODE_CONSTANTS } from 'utils/constants/nodes';
import { NODE_HELPERS } from 'utils/helpers/nodes';

const getClosestDayId = (date, parentNodeId, dayDates) => {
  if (dayDates) {
    const dayIds = dayDates.reduce(
      (acc, { id, value }) =>
        MOMENT_HELPERS.isSame(date, value, 'd') ? [...acc, id] : acc,
      [],
    );
    if (dayIds.includes(parentNodeId)) return parentNodeId;
    if (dayIds.length) return dayIds[0];
  }
  return parentNodeId;
};

const getDate = (dayId, dayDates, defaultDate = FIRST_DATE) => {
  const date = _.find(dayDates, ({ id }) => id === dayId);
  if (date) return date.value;
  return defaultDate;
};

const renderDayOptions = (renderDate, dayDates, otherValue, showEmpty, tabId) =>
  dayDates.reduce(
    (acc, { id, value }, index) => {
      if (MOMENT_HELPERS.isBefore(value, otherValue)) return acc;
      return [
        ...acc,
        {
          value: id,
          children: renderDate(value, index, { alwaysShowIndex: true }),
        },
      ];
    },
    showEmpty ? [{ value: tabId, children: 'No day' }] : [],
  );

const renderDayDateOptions = (renderDate, dayDates) =>
  dayDates.map(({ value }, index) => ({
    value,
    children: renderDate(value, index, {
      startDateRenderFunc: MOMENT_HELPERS.renderDayDate,
    }),
  }));

const renderDayDurationOption = (
  date,
  value,
  renderDuration,
  renderDate,
  includeDate,
  index,
) => {
  const duration = NODE_HELPERS.calculateDayDuration(date, value);
  const renderedDuration = renderDuration(duration);
  if (renderedDuration) {
    if (includeDate) return `${renderedDuration} (${renderDate(value, index)})`;
    return renderedDuration;
  }
  return null;
};

const renderDayDurationOptions = (
  relativeDate,
  dayDates,
  renderDuration,
  renderDate,
  includeDate,
) => {
  if (relativeDate) {
    const options = dayDates.reduce((acc, { value }, index) => {
      const durationValue = NODE_HELPERS.calculateDayDuration(
        relativeDate,
        value,
      ).toISOString();
      const children = renderDayDurationOption(
        relativeDate,
        value,
        renderDuration,
        renderDate,
        includeDate,
        index,
      );
      if (children) return [...acc, { value: durationValue, children }];
      return acc;
    }, []);

    return _.uniqBy(options, 'value');
  }
  return [];
};

const renderDayCount = duration => {
  const days = calculateDayCount(duration) - 1;
  if (days === 0) return 'Same day';
  if (days > 0) {
    const text = pluralizeText('day', days);
    return `${days} ${text} later`;
  }
  return null;
};

const renderNightCount = duration => {
  const nights = calculateDayCount(duration) - 1;
  if (nights === 0) return 'Same day';
  if (nights > 0) {
    const text = pluralizeText('night', nights);
    return `After ${nights} ${text}`;
  }
  return null;
};

const renderDate = (outerOpts = {}) => (value, index, opts = {}) => {
  const { displayDate = null } = outerOpts;
  const {
    alwaysShowIndex = false,
    startDateRenderFunc = MOMENT_HELPERS.renderDayDateShort,
    weekDayRenderFunc = MOMENT_HELPERS.renderDay,
  } = opts;

  let dayIndex;
  let date;
  switch (displayDate) {
    case dateDisplay.startDate:
      dayIndex = alwaysShowIndex ? index : null;
      date = startDateRenderFunc(value);
      break;

    case dateDisplay.weekDay:
      dayIndex = index;
      date = weekDayRenderFunc(value);
      break;

    default:
      dayIndex = index;
      date = '';
      break;
  }
  return NODE_HELPERS.renderDayIndexDate(dayIndex, date);
};

const relativeToStart = (mode, position) =>
  mode === NODE_CONSTANTS.TIME_MODES.relativeStart &&
  position === NODE_CONSTANTS.POSITIONS.end;

const addStartDuration = (mode, position, date, startDuration) => {
  if (date && relativeToStart(mode, position)) {
    return MOMENT_HELPERS.addDurations(date, startDuration);
  }
  return date;
};

const calculateDateTime = (
  mode,
  position,
  duration,
  startDuration,
  parentNodeId,
  dayDates,
) => {
  const date = getDate(parentNodeId, dayDates);
  const parentDateTime = addStartDuration(mode, position, date, startDuration);
  if (parentDateTime) {
    return MOMENT_HELPERS.addDuration(parentDateTime, duration);
  }
  return parentDateTime;
};

const calculateDayCount = duration => (duration ? duration.asDays() + 1 : 0);

export default {
  getClosestDayId,
  getDate,
  renderDayOptions,
  renderDayDateOptions,
  renderDayDurationOptions,
  renderDayCount,
  renderNightCount,
  renderDate,
  relativeToStart,
  addStartDuration,
  calculateDateTime,
  calculateDayCount,
};
