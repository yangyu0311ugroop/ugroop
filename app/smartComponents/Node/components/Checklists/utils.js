import { CLOSED, COMPLETED, OPEN, OUTSTANDING } from 'appConstants';
import { CONSOLE_HELPERS, MISSING_REQUIRED_PROPS } from 'utils/helpers/console';
import { LOGIC_HELPERS } from 'utils/helpers/logic';
import { CHECKITEM, CHECKLIST } from 'utils/modelConstants';

const toggleStatus = (params, props) => {
  const { customData, type, status, ...node } = params;
  const { me } = props;

  let nextStatus;
  let newCustomData;

  switch (type) {
    case CHECKITEM:
      nextStatus = LOGIC_HELPERS.ifElse(
        status === COMPLETED,
        OUTSTANDING,
        COMPLETED,
      );
      newCustomData = LOGIC_HELPERS.ifElse(
        nextStatus === COMPLETED,
        {
          completedAt: new Date().toISOString(),
          completedBy: me,
        },
        {
          completedAt: null,
          completedBy: null,
        },
      );
      break;
    case CHECKLIST:
      nextStatus = LOGIC_HELPERS.ifElse(status === CLOSED, OPEN, CLOSED);
      newCustomData = LOGIC_HELPERS.ifElse(
        nextStatus === CLOSED,
        {
          completedAt: new Date().toISOString(),
          completedBy: me,
        },
        {
          completedAt: null,
          completedBy: null,
        },
      );
      break;
    default:
      CONSOLE_HELPERS.error(
        MISSING_REQUIRED_PROPS,
        'CHECKLIST_HELPERS.toggleStatus: type',
      );
      return null;
  }

  return {
    ...node,
    customData: { ...customData, ...newCustomData },
    type,
    status: nextStatus,
  };
};

export const CHECKLIST_HELPERS = {
  toggleStatus,
};
