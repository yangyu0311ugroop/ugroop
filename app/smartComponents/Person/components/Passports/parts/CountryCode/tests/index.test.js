import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import React from 'react';
import { COUNTRY_LIST_HELPERS } from 'utils/countrylist';
import { VARIANTS } from 'variantsConstants';
import { CountryCode } from '../index';

describe('<CountryCode />', () => {
  let rendered;
  let instance;

  const resaga = {
    setValue: jest.fn(),
    dispatchTo: jest.fn(),
  };

  const props = {
    classes: {},
    resaga,
  };

  beforeEach(() => {
    COUNTRY_LIST_HELPERS.getCountryList = jest.fn(() => []);
    rendered = shallow(<CountryCode {...props} />);
    instance = rendered.instance();
  });

  it('should exists', () => {
    expect(CountryCode).toBeDefined();
  });

  it('should render without exploding', () => {
    expect(rendered.length).toBe(1);
  });

  describe('render()', () => {
    it('should render correctly', () => {
      const snapshot = shallow(<div>{instance.render()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });

    it('should render if variant is default and countryCode is not NONE', () => {
      rendered.setProps({
        countryCode: 'PH',
      });
      const snapshot = shallow(<div>{instance.render()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });

    it('should render correctly if variant is text field', () => {
      rendered.setProps({
        variant: VARIANTS.TEXT_FIELD,
      });
      const snapshot = shallow(<div>{instance.render()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });

    it('should render correctly if variant is text only', () => {
      rendered.setProps({
        variant: VARIANTS.TEXT_ONLY,
      });
      const snapshot = shallow(<div>{instance.render()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });

    it('should render correctly if variant is inline', () => {
      rendered.setProps({
        variant: VARIANTS.INLINE,
      });
      expect(toJSON(rendered)).toMatchSnapshot();
    });

    it('should render correctly if variant is inline and country code is not none', () => {
      rendered.setProps({
        variant: VARIANTS.INLINE,
        countryCode: 'PH',
      });
      expect(toJSON(rendered)).toMatchSnapshot();
    });

    it('should render correctly if variant is text field and country code is not none', () => {
      rendered.setProps({
        variant: VARIANTS.TEXT_FIELD,
        countryCode: 'PH',
      });
      expect(toJSON(rendered)).toMatchSnapshot();
    });
  });
});
