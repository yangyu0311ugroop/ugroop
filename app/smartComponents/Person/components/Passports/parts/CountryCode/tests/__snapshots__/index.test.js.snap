// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CountryCode /> render() should render correctly 1`] = `
<div>
  <WithStyles(EditableSelectForm)
    Typography="H2"
    label="Code of Issuing State/Country Code"
    name="countryCode"
    onSubmit={[Function]}
    options={
      Array [
        Object {
          "children": <WithStyles(JTextClass)
            blue={true}
            italic={true}
          >
            Select none
          </WithStyles(JTextClass)>,
          "value": null,
        },
      ]
    }
    placeholder="Click to specify country code"
    readOnly={false}
    readOnlyPlaceholder="No country code"
    value=""
  />
</div>
`;

exports[`<CountryCode /> render() should render correctly if variant is inline 1`] = `
<WithStyles(EditableSelectForm)
  Typography="H2"
  name="countryCode"
  onSubmit={[Function]}
  options={Array []}
  placeholder="Click to specify country code"
  readOnly={false}
  readOnlyPlaceholder="No country code"
  value=""
/>
`;

exports[`<CountryCode /> render() should render correctly if variant is inline and country code is not none 1`] = `
<WithStyles(EditableSelectForm)
  Typography="H2"
  name="countryCode"
  onSubmit={[Function]}
  options={Array []}
  placeholder="Click to specify country code"
  readOnly={false}
  readOnlyPlaceholder="No country code"
  value="Philippines - PH"
/>
`;

exports[`<CountryCode /> render() should render correctly if variant is text field 1`] = `
<div>
  <Formsy(SelectField)
    autoFocus={false}
    innerRef={[Function]}
    label="Code of Issuing State/Country Code"
    name="countryCode"
    options={
      Array [
        Object {
          "children": <WithStyles(JTextClass)
            blue={true}
            italic={true}
          >
            Select none
          </WithStyles(JTextClass)>,
          "value": null,
        },
      ]
    }
    required={false}
    size="md"
    validationError=""
    validationErrors={Object {}}
    validations={null}
    value="AU"
  />
</div>
`;

exports[`<CountryCode /> render() should render correctly if variant is text field and country code is not none 1`] = `
<Formsy(SelectField)
  autoFocus={false}
  innerRef={[Function]}
  label="Code of Issuing State/Country Code"
  name="countryCode"
  options={
    Array [
      Object {
        "children": <WithStyles(JTextClass)
          blue={true}
          italic={true}
        >
          Select none
        </WithStyles(JTextClass)>,
        "value": null,
      },
    ]
  }
  required={false}
  size="md"
  validationError=""
  validationErrors={Object {}}
  validations={null}
  value="PH"
/>
`;

exports[`<CountryCode /> render() should render correctly if variant is text only 1`] = `<div />`;

exports[`<CountryCode /> render() should render if variant is default and countryCode is not NONE 1`] = `
<div>
  <WithStyles(EditableSelectForm)
    Typography="H2"
    label="Code of Issuing State/Country Code"
    name="countryCode"
    onSubmit={[Function]}
    options={
      Array [
        Object {
          "children": <WithStyles(JTextClass)
            blue={true}
            italic={true}
          >
            Select none
          </WithStyles(JTextClass)>,
          "value": null,
        },
      ]
    }
    placeholder="Click to specify country code"
    readOnly={false}
    readOnlyPlaceholder="No country code"
    value="Philippines - PH"
  />
</div>
`;
